src/mgmt/srv_internal.h

```diff
@@ -95,6 +96,43 @@ int ds_mgmt_destroy_pool(uuid_t pool_uuid, const char *group, uint32_t force);
 void ds_mgmt_hdlr_pool_create(crt_rpc_t *rpc_req);
 void ds_mgmt_hdlr_pool_destroy(crt_rpc_t *rpc_req);
 
+/** srv_query.c */
+
+/* Device health stats from bio_dev_state */
+struct mgmt_bio_health {
+	struct smd_dev_info	*mb_dev_info;
+	uuid_t			 mb_devid;
+	uint64_t		*mb_media_errors;
+	uint64_t		 mb_error_count;
+	uint32_t		 mb_read_errs;
+	uint32_t		 mb_write_errs;
+	uint32_t		 mb_unmap_errs;
+	uint16_t		 mb_temperature;
+	bool			 mb_temp;
+	bool			 mb_spare;
+	bool			 mb_readonly;
+	bool			 mb_reliability;
+	bool			 mb_volatile;
+};
```

Comment:
Is it dup of bio_dev_state? why can't we just use that directly?

Reply:
I will simplify the mgmt_bio_health struct with reference to bio_dev_state to replace most of the struct members. But I think keeping it a different struct for just the mgmt module is a better idea, plus it includes smd_dev_info pointer and also devid

Reply:
Yes, if it's a superset of bio_dev_state, it can include the bio_dev_state to avoid duplications.

Reply:
Looks to me this structure will be used in dPRC buffer? If so, '*mb_dev_info' looks not necessary to me and '*mb_dev_state' should be 'mb_dev_state' (buffer for copying dev state from owner xstream)

Reply:
makes sense, I will remove dev_info and make dev_state not a pointer